demos/adjoint/adjoint_demo.py

```diff
@@ -0,0 +1,342 @@
+# Adjoint Inverse Demo
+# ================================
+#
+# Introduction
+# ------------
+# In this tutorial, we will demonstrate how to perform an inversion to recover the initial temperature field of the mantle using G-ADOPT.
+#
+# This demonstration involves a *twin experiment*, where we assess the performance of the inversion scheme by inverting the initial state of a synthetic reference simulation, known as the "*Reference Twin*". To create this reference twin, we run a forward mantle convection simulation and record all relevant fields (velocity and temperature) at each time step.
+#
+# We have pre-run this simulation and stored the checkpoint file on our servers. These fields serve as benchmarks for evaluating our inverse problem's performance. To download the reference benchmark checkpoint file, click
+# <a href="https://data.gadopt.org/demos/adjoint-demo-checkpoint-state.h5" download> here </a>, or alternatively, execute the following command in a terminal:
+#
+# ```wget https://data.gadopt.org/demos/adjoint-demo-checkpoint-state.h5```
+
+# The fields from the reference simulation are stored under the names "Temperature" and "Velocity". To retrieve the timestepping information, we can use the following code:
+
+# +
+# Importing gadopt
+from gadopt import *
+from gadopt.inverse import *
+checkpoint_filename = "adjoint-demo-checkpoint-state.h5"
```

Comment:
Yes! That's one way of doing this. Or the one suggested for t he tutorial would be downloading the pre-run `.h5` file. and then run it. Note that for the demo I am not checkpointing to disk to avoid mesh issues (ie when number of cores for `forward.py` are different). But `inverse.py` and `taylort_test.py` are exactly as before, and checkpoint to disk.

Reply:
So I don't think we need to get pooch involved here necessarily.